### PR TITLE
fixes for job data map with multiple values, when adding and editing jobs

### PR DESCRIPTION
**Pull request by pablote**
#38

### CODE REVIEW

**Comment by disaacson** [comment]
Thanks for sending this. Here's what I'm seeing:
1) the renumberDataMap function is being somewhat duplicated now
2) the client is sending empty values to the server now and relying on the server to filter them out

Do I misunderstand it?

**Comment by pablote** [comment]
I've just pushed an update.

1) There's no more ambiguity on the js side
2) Regardless of 1), there's always been a chance of empty key/values arriving to the server side. Edit page adds empty key/value inputs that can't be removed and might not get filled, also someone might add fields but never fill them up. The server side cleanup code ensures the job saves correctly.

**Comment by lucidchart** [comment]
Thanks for deduping that function.

A popular rule of thumb in distributed systems is "be strict in what you
send and liberal in what you receive". How about we keep your server side
cleanup, and keep the filter on the client to make sure it sends clean data
as well?

File: admin/app/com/lucidchart/piezo/admin/utils/JobUtils.scala [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.lucidchart.piezo.admin.utils
+
+import org.quartz.JobDetail
+
+object JobUtils {
+
+  def cleanup(job: JobDetail):JobDetail = {
```
Comment (disaacson) [inline admin/app/com/lucidchart/piezo/admin/utils/JobUtils.scala]: scala style is a space after the colon (before the function return type here)

File: admin/app/com/lucidchart/piezo/admin/controllers/Jobs.scala [hunk 2]
```diff
@@ -118,7 +119,7 @@ class Jobs(schedulerFactory: WorkerSchedulerFactory) extends Controller {
       formWithErrors =>
         BadRequest(html.editJob(getJobsByGroup(), formWithErrors, submitNewMessage, formNewAction, false)),
       value => {
-        scheduler.addJob(value, true)
+        scheduler.addJob(JobUtils.cleanup(value), true)
```
Comment (disaacson) [inline admin/app/com/lucidchart/piezo/admin/controllers/Jobs.scala]: mind putting the cleanup on a separate line for clarity?

**Comment by pablote** [comment]
Updated based on code comments, let me know if it looks good, this is the first time I do any scala.

@lucidchart I've heard that one before, although I'm not sure what you mean by "filter on the client". It's just a regular form being posted. Avoiding the empty fields would require more hacky dom manipulating code on the js, as pretty much all the code on that file is. I'm personally more comfortable not adding more confusing code like that, and letting the "liberal in what you receive" do it's thing.

**Comment by radius314** [comment]
:+1:

**Comment by disaacson** [comment]
I'm really sorry for taking forever to review. Thanks for sending it and good work on picking up Scala!

**Pull request merged**